src/neo/Oracle/OraclePolicyContract.cs

```diff
@@ -0,0 +1,230 @@
+        {
+            return new Array(engine.ReferenceCounter, GetOracleValidators(engine.Snapshot).Select(p => (StackItem)p.ToArray()));
+        }
+
+        /// <summary>
+        /// Get current authorized Oracle validator
+        /// </summary>
+        /// <param name="snapshot">snapshot</param>
+        /// <returns>Authorized Oracle validator</returns>
+        public ECPoint[] GetOracleValidators(StoreView snapshot)
+        {
+            ECPoint[] oraclePubKeys = PolicyContract.NEO.GetValidators(snapshot);
+            for (int index = 0; index < oraclePubKeys.Length; index++)
+            {
+                var oraclePubKey = oraclePubKeys[index];
+                StorageKey key = CreateStorageKey(Prefix_Validator, oraclePubKey);
+                ECPoint delegatePubKey = snapshot.Storages.TryGet(key)?.Value.AsSerializable<ECPoint>(1);
+                if (delegatePubKey != null) { oraclePubKeys[index] = delegatePubKey; }
+            }
+            return oraclePubKeys;
+        }
+
+        /// <summary>
+        /// Get number of current authorized Oracle validator
+        /// </summary>
+        /// <param name="engine">VM</param>
+        /// <param name="args">Parameter Array</param>
+        /// <returns>The number of authorized Oracle validator</returns>
+        [ContractMethod(0_03000000, ContractParameterType.Integer)]
+        private StackItem GetOracleValidatorsCount(ApplicationEngine engine, Array args)
+        {
+            return GetOracleValidatorsCount(engine.Snapshot);
+        }
+
+        /// <summary>
+        /// Get number of current authorized Oracle validator
+        /// </summary>
+        /// <param name="snapshot">snapshot</param>
+        /// <returns>The number of authorized Oracle validator</returns>
+        public BigInteger GetOracleValidatorsCount(StoreView snapshot)
+        {
+            return GetOracleValidators(snapshot).Length;
+        }
+
+        /// <summary>
+        /// Create a Oracle multisignature address
+        /// </summary>
+        /// <param name="snapshot">snapshot</param>
+        /// <returns>Oracle multisignature address</returns>
+        public UInt160 GetOracleMultiSigAddress(StoreView snapshot)
+        {
+            ECPoint[] validators = GetOracleValidators(snapshot);
+            return Contract.CreateMultiSigRedeemScript(validators.Length - (validators.Length - 1) / 3, validators).ToScriptHash();
+        }
+
+        /// <summary>
+        /// Set timeout
+        /// </summary>
+        /// <param name="engine">VM</param>
+        /// <param name="args">Parameter Array</param>
+        /// <returns>Returns true if the execution is successful, otherwise returns false</returns>
+        [ContractMethod(0_03000000, ContractParameterType.Boolean, ParameterTypes = new[] { ContractParameterType.Integer }, ParameterNames = new[] { "TimeOutMilliSeconds" })]
+        private StackItem SetTimeOutMilliSeconds(ApplicationEngine engine, Array args)
+        {
```

Comment:
It's better to move the configuration to a specific class? SetConfig and convert this array into a class?

```
OraclePolicyconfig
{
public int Timeout{get;set;}
}
```

I think that the configuration could be very dependant of the protocol, so this method it's very generic.

Reply:
By using the protocol type as a parameter, different timeouts are returned for different types?
If I move it into a specific class, I can implement it with a map, can I?
Does the cost need to be set separately based on the protocol type?

Reply:
My worry it's about that different protocols can have different configurations.

Reply:
> Does the cost need to be set separately based on the protocol type?

I think not at this point.

Reply:
I think that if you want to set different timeouts for each different protocol, you can add the corresponding case implementation directly in the method.You can pass the protocol type as a parameter, and then judge to return a different value.It doesn't feel much need to use a separate class
@shargon

Reply:
Or as you said, we set different configuration classes according to different protocols, such as HttpConfig \ HttpsConfig, and then I add the corresponding Set / Get method in OraclePolicy, such as SetHttpConfig / GetHttpConfig, how about this?